spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -552,6 +556,10 @@ protected void doShutdown() {
 				this.hasStopped = true;
 			}
 		}
+		if (!this.taskSchedulerSet && this.taskScheduler != null) {
+			((ThreadPoolTaskScheduler) this.taskScheduler).shutdown();
+			this.taskScheduler = null;
```

Comment:
Well, with the `squash` GH doesn't see changes and don't notify via email.
I have missed this change therefore :cry: